### PR TITLE
feat: (wip)track discussions

### PR DESCRIPTION
**Pull request by claudia-codes**
This is a work in progress (functionality NOT finished)

This ticket closes https://github.com/nearform/github-snooze-chrome-extension/issues/402

Context: Add the ability to support Discussions (not PR discussions or Team Discussions) 

Steps to test:
- Follow all the steps from readme and use [Classic PAT](https://github.com/settings/tokens/new) NOT [Fine-grained PAT](https://github.com/settings/personal-access-tokens/new).
- Navigate to a repository that has "Discussions" enabled, for example https://github.com/nearform/graphql-hooks/discussions/789
Screenshot 2023-01-13 at 18 22 51
- Create snooze 
Screenshot 2023-01-13 at 13 54 48

Known issues:
- Github doesn't allow graphql requests using [Fine-grained PAT](https://github.com/settings/personal-access-tokens/new)
- The PAT token doesn't seem to load in time and request appear unauthenticated (in index.jsx, createClient, L15)
- urql uses hooks and adding/updating is handled all in a callback, there needs to be a way to wait for executeQuery to retreive data

### CODE REVIEW

**Comment by lasmil** [comment]
Proof that graphql does not allow fine grained tokens yet:
image
(from here: https://docs.github.com/en/graphql/guides/forming-calls-with-graphql )

File: README.md [hunk 1]
```diff
@@ -32,13 +32,15 @@ You will need to:
...
     - `Metadata` (should be set automatically when you set it for `Issues` or `Pull requests`)
+    - `Discussions` - currently not available with fine-grained PATs
 
 #### Generating a Classic PAT
 You will need to:
 - [open the Classic PAT generation page](https://github.com/settings/tokens/new)
 - select the following scopes:
     - `repo`: used for getting the repository information, such as last update date time on issues and pull requests
     - `read:user`: used for retrieving the current logged in user information _(GitHub ID and username)_
+    - `write:discussions`: used to have access to the discussions API
```
Comment (simoneb) [inline README.md:43]: why write?

File: src/api/github.js [hunk 2]
```diff
@@ -21,6 +24,39 @@ export const getUserByPat = async token => {
...
+        return result;
+      }
+      return {
+        updated_at: result?.repository.discussion.updatedAt
+      }
+    } catch (error) {
+      return {
+        error
+      }
+    }
```
Comment (simoneb) [inline src/api/github.js:57]: refactor into a function please for readability and leverage async await

File: src/graphql/client.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      const { pat } = await readFromLocalStorage([SK_PAT, SK_USER])
+
+      if (!pat) {
+          throw Error('PAT not available.') 
+      }
+
+      return { token: pat }
+  }
+
+  return null;
```
Comment (simoneb) [inline src/graphql/client.js:18]: invert conditional to stick to early return pattern, also don't return nulls

File: src/graphql/client.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+};
+
+const addAuthToOperation = ({ authState, operation }) => {
+  if (!authState || !authState.token) {
```
Comment (simoneb) [inline src/graphql/client.js:22]: minor: we can probably use conditional chaining operator here

**Review by simoneb — COMMENTED** [review]
nothing major or blocking, but could use some minor tweaks before being tested out please

File: README.md (hunk 1 above)
Comment (lasmil) [reply README.md]: Yepp, sorry about that. I had both checkboxes checked when I created the classic PAT. It does indeed work with read:discussions and there is no need for write.

File: src/api/github.js (hunk 2 above)
Comment (lasmil) [reply src/api/github.js]: Function extracted.

File: src/graphql/client.js (hunk 3 above)
Comment (lasmil) [reply src/graphql/client.js]: Yepp. Based on https://formidable.com/open-source/urql/docs/advanced/authentication/ the **getAuth** was returning null. We know that the return value of this is used in the **addAuthToOperation**() function. To not return null, we can return an object: **{ token: null }** instead. Both options work in the **addAuthToOperation**() function, but your suggestion avoid using nulls.
PR updated.

File: src/graphql/client.js (hunk 4 above)
Comment (lasmil) [reply src/graphql/client.js]: Yepp. Done

**Review by simoneb — APPROVED** [review]

**Pull request merged**